lib/pbench/test/unit/server/query_apis/test_datasets_list.py

```diff
@@ -98,37 +66,8 @@ def test_missing_keys(self, client, server_config, pbench_token, keys):
         however, Pbench will silently ignore any additional keys that are
         specified.
        """
-        response = client.post(
-            f"{server_config.rest_uri}/datasets/list",
-            headers={"Authorization": "Bearer " + pbench_token},
-            json=keys,
-        )
-        assert response.status_code == HTTPStatus.BAD_REQUEST
-        missing = [k for k in ("controller", "start", "end") if k not in keys]
-        assert (
-            response.json.get("message")
-            == f"Missing required parameters: {','.join(missing)}"
-        )
-
-    def test_bad_dates(self, client, server_config, pbench_token):
-        """
-        Test behavior when a bad date string is given
-        """
-        response = client.post(
-            f"{server_config.rest_uri}/datasets/list",
-            headers={"Authorization": "Bearer " + pbench_token},
-            json={
-                "user": "drb",
-                "controller": "dbutenho.csb",
-                "start": "2020-12",
-                "end": "2020-19",
-            },
-        )
-        assert response.status_code == HTTPStatus.BAD_REQUEST
-        assert (
-            response.json.get("message")
-            == "Value '2020-19' (str) cannot be parsed as a date/time string"
-        )
+        self.required_keys = ("controller", "start", "end")
```

Comment:
This can be dynamically configured from the schema!

Reply:
This would need creating an object, like `datasetlist = DatasetsList(PbenchServerConfig, Logger)` and then we can check the schema that belongs to that object right?

Reply:
Yes, it would. That's a very "unit test" way to do things, although we've actually been driving the API tests through HTTP-level mocking. Arguably we should be doing both, although what we're doing (especially in the cases where we're using a live DB) are really more "functional" than unit.